co2calculator/calculate.py

```diff
@@ -178,163 +252,100 @@ def apply_detour(distance: Kilometer, transportation_mode: str) -> Kilometer:
     return distance_with_detour
 
 
-def calc_co2_bus(
-    distance: Kilometer = None,
-    stops: list = None,
-    size: str = None,
-    fuel_type: str = None,
-    occupancy: int = None,
-    vehicle_range: str = None,
-) -> Tuple[Kilogram, Kilometer]:
+class Bus(BaseModel):
+    """Vehicle type Bus"""
+
+    @enum.unique
+    class Size(str, enum.Enum):
+        """Sizes for busses"""
+
+        MEDIUM = "medium"
+        LARGE = "large"
+        AVERAGE = "average"
+
+    class FuelType(str, enum.Enum):
+        """Fuel types for busses"""
+
+        DIESEL = "diesel"
+
+    class BusRange(str, enum.Enum):
+        """Range types for busses"""
+
+        LONG_DISTANCE = "long-distance"
+
+    size: Size
+    fuel_type: FuelType
+    occupancy: int  # TODO: Validate to 20, 50, 80 or 100
+    vehicle_range: BusRange
+
+
+class BusEmissionRequest(EmissionRequest):
+    """Request model for bus emission calculation"""
+
+    vehicle: Bus
+
+
+def calc_co2_bus(request: BusEmissionRequest) -> float:
```

Comment:
Here you see the effect: 

Arguments? Simple!

Validation? Already done!

Access of values? Easy!